src/sentry/eventstore/snuba/backend.py

```diff
@@ -107,6 +107,7 @@ def __get_events(
             and filter.project_ids
             and len(filter.event_ids) * len(filter.project_ids) < min(limit, NODESTORE_LIMIT)
             and offset == 0
+            and should_bind_nodes
```

Comment:
@jan-auer I just added this check rather than adding the separate `get_unfetched_event_by_id` method since it sounds like you plan to switch to bulk loading soon anyway. Does that sound reasonable?

Reply:
I'm not sure about bulk loading since that would require us to do something different than what the current methods do. Regardless, I think checking the flag here makes sense and unblocks me 👍

Reply:
I think you can still use this method in future, just pass in a list of OR'ed conditions to the filter